### PR TITLE
Refactored model server test suite and added new connector related test cases

### PR DESCRIPTION
**Pull request by harshbafna**
## Description

 Refactors existing UT to use normal port for testing model server and added a new test class to validate all supported port combinations.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Feature/Issue validation/testing

Refer CI logs.

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?

### CODE REVIEW

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    @Test
+    public void testSSLInferenceAddress() {
+        Connector conn =
+                Connector.parse("https://127.0.0.1:45245", ConnectorType.INFERENCE_CONNECTOR);
+
+        Assert.assertEquals(conn.getSocketType(), "tcp");
+        Assert.assertEquals(conn.getSocketPath(), "45245");
+        Assert.assertFalse(conn.isManagement());
```
Comment (chauhang) [inline frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:39]: @harshbafna Looks like copy/paste error in the Inference port check

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        Assert.assertFalse(conn.isSsl());
+        Assert.assertFalse(conn.isUds());
+    }
+
+    @Test
+    public void testSSLInferenceAddress() {
+        Connector conn =
+                Connector.parse("https://127.0.0.1:45245", ConnectorType.INFERENCE_CONNECTOR);
```
Comment (chauhang) [inline frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:35]: @harshbafna Shouldn't this  be a different port number vs being same as the Management port in the previous test?

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        Assert.assertEquals(conn.getSocketPath(), "45245");
+        Assert.assertTrue(conn.isManagement());
+        Assert.assertFalse(conn.isSsl());
+        Assert.assertFalse(conn.isUds());
+    }
+
+    @Test
+    public void testValidInferenceAddress() {
+        Connector conn =
+                Connector.parse("http://127.0.0.1:45245", ConnectorType.INFERENCE_CONNECTOR);
```
Comment (chauhang) [inline frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:23]: @harshbafna Shouldn't this be a different port number vs being same as the Management port in the previous test?

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+        Assert.assertEquals(conn.getSocketType(), "tcp");
+        Assert.assertEquals(conn.getSocketPath(), "45245");
+        Assert.assertFalse(conn.isManagement());
```
Comment (chauhang) [inline frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:27]: @harshbafna Looks like copy/paste error in the Inference port check

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
+package org.pytorch.serve.util;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class ConnectorTest {
```
Comment (chauhang) [inline frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:6]: @harshbafna Please add tests for the new Metrics port checks as well

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        Assert.assertFalse(conn.isSsl());
+        Assert.assertFalse(conn.isUds());
+    }
+
+    @Test
+    public void testDefaultInferenceAddress() {
+        Connector conn = Connector.parse("http://127.0.0.1", ConnectorType.INFERENCE_CONNECTOR);
+        Assert.assertEquals(conn.getSocketType(), "tcp");
+        Assert.assertEquals(conn.getSocketPath(), "80");
+        Assert.assertFalse(conn.isManagement());
```
Comment (chauhang) [inline frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:93]: @harshbafna Looks like copy/paste error in the Inference address check

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        Assert.assertTrue(conn.isSsl());
+        Assert.assertFalse(conn.isUds());
+    }
+
+    @Test
+    public void testDefaultSSLInferenceAddress() {
+        Connector conn = Connector.parse("https://127.0.0.1", ConnectorType.INFERENCE_CONNECTOR);
+        Assert.assertEquals(conn.getSocketType(), "tcp");
+        Assert.assertEquals(conn.getSocketPath(), "443");
+        Assert.assertFalse(conn.isManagement());
```
Comment (chauhang) [inline frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:113]: @harshbafna Looks like copy/paste error in the Inference address check

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java (hunk 2 above)
Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:46]: @chauhang: These test cases don't block any port and thus we are free to use the same port number in every test case.

The idea here is to check whether the connector object is getting created properly as per the input provided. This allows us to remove the dependency of ModelServerTest suite on the `Unix` socket for testing.

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java (hunk 5 above)
Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:6]: Done.

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java (hunk 3 above)
Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:23]: @chauhang: These test cases don't block any port and thus we are free to use the same port number in every test case.

The idea here is to check whether the connector object is getting created properly as per the input provided. This allows us to remove the dependency of ModelServerTest suite on the Unix socket for testing.

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java (hunk 4 above)
Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:27]: @chauhang: It's a negative assertion i.e. `assertFalse`

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java (hunk 1 above)
Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:50]: @chauhang: It's a negative assertion i.e. assertFalse

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java (hunk 6 above)
Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:128]: @chauhang: It's a negative assertion i.e. assertFalse

File: frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java (hunk 7 above)
Comment (harshbafna) [reply frontend/server/src/test/java/org/pytorch/serve/util/ConnectorTest.java:158]: @chauhang: It's a negative assertion i.e. assertFalse

**Comment by harshbafna** [comment]
> @harshbafna Please see comments inline and add new tests for the Metrics port.
> 
> Further please clarify if this will solve the issues for testing on Windows platform?

@chauhang: I have added the metric connector related test cases and responded to other comments. This PR should solve the problem related to the Unix socket being used in `ModelServerTest` for starting torchserve and validate it using the newly added test file `ConnectorTest.java`

**Comment by harshbafna** [comment]
This has been merged with Windows Native Support PR (#662). Closing this.

**Pull request closed without merging**